File: 3765.complete-prime-number.py
#
# @lc app=leetcode id=3765 lang=python3
#
# [3765] Complete Prime Number
#

# @lc code=start
class Solution:
    def completePrime(self, num: int) -> bool:
        def is_prime(n):
            if n < 2:
                return False
            if n == 2:
                return True
            if n % 2 == 0:
                return False
            i = 3
            while i * i <= n:
                if n % i == 0:
                    return False
                i += 2
            return True

        s = str(num)
        n = len(s)
        # Check all prefixes
        for i in range(1, n+1):
            prefix = int(s[:i])
            if not is_prime(prefix):
                return False
        # Check all suffixes
        for i in range(n):
            suffix = int(s[i:])
            if not is_prime(suffix):
                return False
        return True
# @lc code=end

File: 3765.complete-prime-number.java
// @lc app=leetcode id=3765 lang=java
//
// [3765] Complete Prime Number
//
// @lc code=start
class Solution {
    public boolean completePrime(int num) {
        String s = String.valueOf(num);
        int n = s.length();
        for (int i = 1; i <= n; ++i) {
            int prefix = Integer.parseInt(s.substring(0, i));
            int suffix = Integer.parseInt(s.substring(n - i));
            if (!isPrime(prefix) || !isPrime(suffix)) {
                return false;
            }
        }
        return true;
    }
    private boolean isPrime(int x) {
        if (x < 2) return false;
        if (x == 2) return true;
        if (x % 2 == 0) return false;
        for (int i = 3; i * i <= x; i += 2) {
            if (x % i == 0) return false;
        }
        return true;
    }
}
// @lc code=end